### DISCUSSION TITLE
Netbox on RedHat - Upgrade Script

### DISCUSSION THREAD

**Discussion by berkeleycowboy** [netbox-community/netbox#12948] [Help Wanted!]
When performing the Netbox instillation performing the "Create a Super User"  the following command is executed:

 ~]$ source /opt/netbox/venv/bin/activate
-bash: /opt/netbox/venv/bin/activate: Permission denied
 ~]$ 
 ~]$ sudo source /opt/netbox/venv/bin/activate
[sudo] password for weikeld: 
sudo: source: command not found

Do you have recommendations about how to resolve the above errors.

**Comment by pobradovic08** [comment]
`source` requires existing shell (that `sudo` by default does not start).

Do you have access to files in `/opt/netbox` ?
Can you run `/opt/netbox/venv/bin/python3 manage.py createsuperuser`?

**Comment by berkeleycowboy** [comment]
I did figure out what was wrong and wanted to share so that others could avoid the issues I faced.  

The Netbox configuration section of the Installation & upgrade manual appears to be missing one configuration change.    After the Secret_Key is generated it needs to be inputted into the configuration.py located in the /opt/netbox/netbox/netbox/ folder.  

The change is illustrated in the video

**Reply by abhi1693** [reply]
This is not missing from the docs at all. It's clearly stated that you need to update your configuration.py file with the required parameters among which is the SECRET_KEY

https://docs.netbox.dev/en/stable/installation/3-netbox/

**Comment by berkeleycowboy** [comment]
I reread the Netbox and installation and you are right it does state the following four are required elements for new installations.  
* ALLOWED_HOSTS
* DATABASE
*  REDIS
*  SECRET_KEY

I see now how the the configuration section was intended now.  The SECRET_KEY section appeared to be ONLY associated with generating the secret key.  

I wish I had seen the Table of contents on the right hand side of the page because its clear what was intended.  

Thank you.
